modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1116,6 +1116,20 @@ public String escapeText(String input) {
                         .replace("\"", "\\\""));
     }
 
+    /**
+     * This method escapes text to be used in a single quoted string
+     * @param input the input string
+     * @return the escaped string
+     */
+    public String escapeTextInSingleQuotes(String input) {
+        if (input == null) {
+            return null;
+        }
+
+        return escapeText(input).replace("'", "\\'");
+    }
+
+
```

Comment:
in abstract typescript codegen, it overrides escapeText to replace `'` with `\\'`

https://github.com/OpenAPITools/openapi-generator/blob/e914c4098b44ac734d3dd21300d433f6cb2974a5/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L1068

what about follow the same without introducing `escapeTextInSingleQuotes`?

Reply:
There are several issues with escaping single quotes by default that would also affect other generators:
- Some languages (e.g., PHP) treat `'\"'` as a string containing a backslash and a double quote
- `'` is often used in text like parameter descriptions and would be escaped in comments, which leads to doc blocks like this:
![image](https://github.com/user-attachments/assets/c40ea729-f5f1-4ecc-9ac4-008d9d8a5904)
- Escaping a single quote is illegal in some formats (e.g., JSON)

Ideally, the context a text is used in (e.g., JSON, single-quoted string, double-quoted string or comment) could be taken into account when a string is escaped, either by making them individual methods or by adding a parameter for that.

Reply:
thanks for the explanation. let me think about it and review accordingly